MediaBrowser.Model/Extensions/ContainerHelper.cs

```diff
@@ -0,0 +1,142 @@
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(string? profileContainers, bool isNegativeList, string? inputContainer)
+    {
+        if (string.IsNullOrEmpty(inputContainer))
+        {
+            return isNegativeList;
+        }
+
+        return ContainsContainer(profileContainers, isNegativeList, inputContainer.AsSpan());
+    }
+
+    /// <summary>
+    /// Compares two containers, returning <paramref name="isNegativeList"/> if an item in <paramref name="inputContainer"/>
+    /// does not exist in <paramref name="profileContainers"/>.
+    /// </summary>
+    /// <param name="profileContainers">The comma-delimited string being searched.</param>
+    /// <param name="isNegativeList">The boolean result to return if a match is not found.</param>
+    /// <param name="inputContainer">The comma-delimited string being matched.</param>
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(string? profileContainers, bool isNegativeList, ReadOnlySpan<char> inputContainer)
+    {
+        if (string.IsNullOrEmpty(profileContainers))
+        {
+            // Empty profiles always support all containers/codecs.
+            return true;
+        }
+
+        var allInputContainers = inputContainer.Split(',');
+        var allProfileContainers = profileContainers.SpanSplit(',');
+        foreach (var container in allInputContainers)
+        {
+            foreach (var profile in allProfileContainers)
+            {
+                if (MemoryExtensions.Equals(profile, container, StringComparison.OrdinalIgnoreCase))
+                {
+                    return !isNegativeList;
+                }
+            }
+        }
+
+        return isNegativeList;
+    }
+
+    /// <summary>
+    /// Compares two containers, returning <paramref name="isNegativeList"/> if an item in <paramref name="inputContainer"/>
+    /// does not exist in <paramref name="profileContainers"/>.
+    /// </summary>
+    /// <param name="profileContainers">The profile containers being matched searched.</param>
+    /// <param name="isNegativeList">The boolean result to return if a match is not found.</param>
+    /// <param name="inputContainer">The comma-delimited string being matched.</param>
+    /// <returns>The result of the operation.</returns>
+    public static bool ContainsContainer(IReadOnlyList<string>? profileContainers, bool isNegativeList, string inputContainer)
+    {
+        if (profileContainers is null)
+        {
+            // Empty profiles always support all containers/codecs.
+            return true;
+        }
+
+        var allInputContainers = inputContainer.Split(',');
+        foreach (var container in allInputContainers)
+        {
+            foreach (var profile in profileContainers)
+            {
+                if (MemoryExtensions.Equals(profile, container, StringComparison.OrdinalIgnoreCase))
```

Comment:
Both are strings, no?
```suggestion
                if (string.Equals(profile, container, StringComparison.OrdinalIgnoreCase))
```